### PR TITLE
[aspnetcore] Feature/cleanup

### PR DESCRIPTION
**Pull request by A-Joshi**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`~~, `3.4.x`, `4.0.x`~~. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Mostly cosmetic code refactoring to make the code more readable - as per [2285](https://github.com/OpenAPITools/openapi-generator/issues/2285)

- Make classModifier, operationModifier and buildTarget enums
- Enhance CliOption class to hold value - it reduces boilerplat code.
- Add ASP.Net Core 2.2 (no changes to generated code - since it is pretty much backward compatible with 2.1).

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/CliOption.java [hunk 1]
```diff
@@ -73,6 +74,15 @@ public CliOption defaultValue(String defaultValue) {
         return this;
     }
 
+    public String getOptValue() { return this.optValue;}
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/CliOption.java]: @A-Joshi minor suggestion: add a line break here.

**Comment by wing328** [comment]
cc @jimschubert as this PR enhances the CliOption class.

**Comment by wing328** [comment]
@A-Joshi thanks for the PR. Just a heads up that there's another project to add a ASP.NET core 3.0 generator: https://github.com/OpenAPITools/openapi-generator/projects/9

**Comment by A-Joshi** [comment]
Thanks will reach out - it should be easy to add in the 3.0 support as a
folder and target to the current setup so we don't have yet another
almost the same generator

**Comment by wing328** [comment]
I've not tried 3.0 myself. Is that very different when compared with the 2.x version?

For other generators (e.g. Swift), we deal with major version upgrade with a new generator to make it easier to maintain.

For ASP.NET CORE, I do not have a preference as I'm not familiar with the framework.

**Comment by A-Joshi** [comment]
So far the changes are mostly for performance and some thirdparty
packages have been removed. I'll reach out to see what would work best.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/CliOption.java (hunk 1 above)
Comment (A-Joshi) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/CliOption.java]: Done.

**Comment by wing328** [comment]
Looks like a rebase fails and this PR now contains commits not authored by you.

I wonder if you can create a new branch based on the latest master and cherry-pick your commits.

**Comment by jimschubert** [comment]
Changes look good.  I've rebased onto my own branch to evaluate only the changes related to this PR. Feel free to pull/branch off that and reopen a PR:  https://github.com/jimschubert/openapi-generator/tree/ihsmarkitoss-feature/cleanup. Probably best to have @A-Joshi double-check just to make sure I didn't accidentally remove any intended work during the rebase.

**Comment by jimschubert** [comment]
I'm closing this in favor of #2506, which is rebased against master and includes original contributor information during the rebase.

**Comment by A-Joshi** [comment]
Thanks have been traveling abroad will look as soon as I am back this week

**Pull request closed without merging**